src/plugins/auto_batch/src/compiled_model.cpp

```diff
@@ -154,93 +134,86 @@ std::pair<CompiledModel::WorkerInferRequest&, int> CompiledModel::GetWorkerInfer
             }
         });
     }
-    return {*m_worker_requests.back(), static_cast<int>(batch_id)};
+    return {m_worker_requests.back(), static_cast<int>(batch_id)};
 }
 
-InferenceEngine::IInferRequestInternal::Ptr CompiledModel::CreateInferRequest() {
-    if (!m_model_with_batch) {
-        auto res = m_model_without_batch->CreateInferRequest();
-        res->setPointerToExecutableNetworkInternal(shared_from_this());
-        res->setPointerToSo(m_model_without_batch._so);
-        _so = m_model_without_batch._so;
-        return res;
+std::shared_ptr<ov::IAsyncInferRequest> CompiledModel::create_infer_request() const {
+    if (!m_compiled_model_with_batch) {
+        return m_compiled_model_without_batch->create_infer_request();
```

Comment:
probably you will have life cycle issue here, for example, for batch_infer_request.get_tensor() you need to make sure the tensor holds reference for both batch and underlying hardware plugin so.

Reply:
I suppose it should be fixed by overriding InferRequest::get_tensor and adding extra so here?